handsontable/src/helpers/dom/element.js

```diff
@@ -969,6 +969,28 @@ export function setCaretPosition(element, pos, endPos) {
 
 let cachedScrollbarWidth;
 
+/**
+ * Returns the fractional scaling compensation for scrollbar width calculation.
+ *
+ * @param {Document} rootDocument The onwer of the document.
+ * @returns {number} The compensation for the scrollbar width, when the device pixel ratio is not an integer.
+ */
+// eslint-disable-next-line no-restricted-globals
+export function getScrollbarFractionalScalingCompensation(rootDocument = document) {
+  const rootWindow = rootDocument.defaultView;
+  const isWindows = rootWindow.navigator.userAgent.toLowerCase().includes('windows');
+
+  if (!isWindows) {
+    return 0;
+  }
+
+  const devicePixelRatio = rootWindow.devicePixelRatio || 1;
+
+  // On Windows, fractional scaling makes the scrollbar wider to compensate for the anti-aliasing.
+  // This is a workaround to calculate the correct scrollbar width.
+  return Number.isInteger(devicePixelRatio) ? 0 : 2;
```

Comment:
Just guessing .. `Math.ceil` doesn't solve the case in a more general way?

Reply:
I was going to use `Math.ceil`, but it would up the value by `1`, and the difference that seems to be happening is `2px` (I'm guessing anti-aliasing on both sides of the scrollbar?), so it wouldn't be sufficient.